kernel/enter.c

```diff
@@ -938,6 +948,16 @@ int myst_enter_kernel(myst_kernel_args_t* args)
         free(process);
         process = NULL;
 
+        if (create_appenv_ret == 0)
```

Comment:
Suggest doing the clean up in `myst_create_appenv` instead (only when successful)

Reply:
We use the value after the call to that function, so the free needs to be after the exec statement in enter.c